src/main/java/com/vinyl/repository/TokenRepository.java

```diff
@@ -10,4 +10,5 @@
 public interface TokenRepository extends CrudRepository<Token, Integer>{
 
 	Token findFirstByUserOrderByValidUntilDesc(User user);
+	Token findByUserId(Integer userId);
```

Comment:
the user and token has a 1 to many relationship, if you try to find tokens by a user id you will get by a list (not a single user). I think you can just use the existing method findFirstByUserOrderByValidUntilDesc to check if the token exists

Reply:
done